### ISSUE TITLE
Installation of HANA SPS04 Rev 46 fails due to permission issues 

### ISSUE THREAD

**Issue by newkit** [redhat-sap/sap-hana-deployment#14]
Platform: PPC, not sure about x86

Here is the same issue as I see when running the ansible role:

```
[root@hostname SAP_HANA_DATABASE]# ./hdblcm --configfile=/tmp/ansible.d6m61chehanaconfig/configfile.cfg -b


SAP HANA Lifecycle Management - SAP HANA Database 2.00.046.00.1581325702
************************************************************************


Scanning software locations...
Detected components:
    SAP HANA Database (2.00.046.00.1581325702) in /tmp/ansible.2b3pdnaxdci/SAP_HANA_DATABASE/server
Running in batch mode
  The following software installation files are missing execute permissions:
    /tmp/ansible.2b3pdnaxdci
  Ensure that no permissions are changed by the umask when extracting or copying the folders.

```

Can be fixed by:

` chmod -R a+rwx  /tmp/ansible.2b3pdnaxdci`

**Comment by newkit** [comment]
As discussed with Ricardo, this might be an issue bc of permissions under /tmp.
One thing might be to choose a different location where to extract the HANA.SAR file,
maybe making it configurable by the user.

**Comment by newkit** [comment]
FYI, same on x86:

ASK [sap_hana_deployment : Install SAP HANA] **********************************
fatal: [lu0548.wdf.sap.corp]: FAILED! => {"changed": true, "cmd": ["./hdblcm", "--configfile=/tmp/ansible.plb3luvvhanaconfig/configfile.cfg", "-b"], "delta": "0:00:00.792262", "end": "2020-04-17 13:10:11.162470", "msg": "non-zero return code", "rc": 1, "start": "2020-04-17 13:10:10.370208", "stderr": "Running in batch mode\n  The following software installation files are missing execute permissions:\n    /tmp/ansible.gf1xao34dci\n  Ensure that no permissions are changed by the umask when extracting or copying the folders.", "stderr_lines": ["Running in batch mode", "  The following software installation files are missing execute permissions:", "    /tmp/ansible.gf1xao34dci", "  Ensure that no permissions are changed by the umask when extracting or copying the folders."], "stdout": "\n\nSAP HANA Lifecycle Management - SAP HANA Database 2.00.046.00.1581325702\n************************************************************************\n\n\nScanning software locations...\nDetected components:\n    SAP HANA Database (2.00.046.00.1581325702) in /tmp/ansible.gf1xao34dci/SAP_HANA_DATABASE/server\nLog file written to '/var/tmp/hdb_NKT_hdblcm_install_2020-04-17_13.10.10/hdblcm.log' on host 'lu0548'.", "stdout_lines": ["", "", "SAP HANA Lifecycle Management - SAP HANA Database 2.00.046.00.1581325702", "************************************************************************", "", "", "Scanning software locations...", "Detected components:", "    SAP HANA Database (2.00.046.00.1581325702) in /tmp/ansible.gf1xao34dci/SAP_HANA_DATABASE/server", "Log file written to '/var/tmp/hdb_NKT_hdblcm_install_2020-04-17_13.10.10/hdblcm.log' on host 'lu0548'."]}

**Comment by rhmk** [comment]
doublechek if you have mounted /tmp on tmpfs.  My /tmp us part of / and the tmpfs stuff goes here:
# grep tmp /etc/mtab
devtmpfs /dev devtmpfs rw,seclabel,nosuid,size=10089004k,nr_inodes=2522251,mode=755 0 0
tmpfs /dev/shm tmpfs rw,seclabel,nosuid,nodev 0 0
tmpfs /run tmpfs rw,seclabel,nosuid,nodev,mode=755 0 0
tmpfs /sys/fs/cgroup tmpfs ro,seclabel,nosuid,nodev,noexec,mode=755 0 0
tmpfs /run/user/42 tmpfs rw,seclabel,nosuid,nodev,relatime,size=2023612k,mode=700,uid=42,gid=42 0 0
tmpfs /run/user/15766 tmpfs rw,seclabel,nosuid,nodev,relatime,size=2023612k,mode=700,uid=15766,gid=15766 0 0

If your system setup has mounted /tmp on tmpfs and your noexec or nosuid  flag is set this could cause the problem.  Otherwise /tmp should behave as any other directory

**Comment by makentenza** [comment]
@newkit @rhmk I don't think this issue is related to this role in particular since this is complaining on the installation source files, which is provided by the user. You can always add execution flag to these files before using the role

**Comment by tlinnet** [comment]
This was performed on Ubuntu 22.04 running on VirtualBox

```bash
# Fix HANA scripts are not compatible with dash shell. /bin/sh is a softlink to /usr/bin/dash. 
sudo mv /bin/sh /bin/sh.orig
sudo ln -s /bin/bash /bin/sh
# Install missing packages. # See which version installed of libstdc++6: 12.1
sudo apt install libltdl7 zip unzip libstdc++6
apt-cache policy libstdc++6
# We need to create the group sapsys with id 79
# Else you get error: Ensure that no permissions are changed by the umask when extracting or copying the folders.
# First verify if already exists
getent group sapsys
getent group 79
# Then we create the group and set it as primary group
sudo groupadd -g 79 sapsys
sudo usermod -a -G sapsys vboxuserhana
sudo usermod -g sapsys vboxuserhana
# Login again and check primary group
su vboxuserhana
id
# Unpack and install
mkdir hanainstall
mv hxe.tgz hanainstall
cd hanainstall
tar -xvzf hxe.tgz
# Fix issue pin to libstdc++6-6.2 net in setup_hxe.sh
sed -i 's/libstdc++6-6.2/libstdc++6/g' ./setup_hxe.sh
# Then install
sudo ./setup_hxe.sh
```

The group id of sapsys=79, was found by following:
https://support.huaweicloud.com/intl/en-us/ugs-saphana/saphana_02_0045.html

Instead of ./setup_hxe.sh, then do
```bash
cd ./HANA_EXPRESS_20/DATA_UNITS/HDB_SERVER_LINUX_X86_64
sudo ./hdblcm --ignore=check_signature_file
sudo chmod -R 777 /hana
sudo ./hdblcm --ignore=check_signature_file
chmod -R a+rwx /home/vboxuserhana
```
